src/main/java/tech/allegro/wildsnake/model/ProductCategoryDomain.java

```diff
@@ -0,0 +1,54 @@
+package tech.allegro.wildsnake.model;
+
+import tech.allegro.wildsnake.product.model.Product;
+import tech.allegro.wildsnake.productCategory.model.ProductCategory;
+
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+
+public class ProductCategoryDomain {
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private long id;
```

Comment:
Consider use UUID type for uniq category id. To create random id you can use  idOne = UUID.randomUUID()